src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -76,42 +76,41 @@ export const MultiSelect = chakra(
         onClear,
         onMixedStateChange,
         selectedItems,
-        width = "full",
+        multiSelectWidth = "full",
         ...rest
       } = props;
 
       // Create a ref to hold a reference to the accordian button, enabling us to programmatically focus it.
       const accordianButtonRef: React.RefObject<HTMLDivElement> =
         useRef<HTMLDivElement>();
 
-      // Control the open or closed state of the MultiSelect.
-      const DEFAULT_ITEMS_LIST_HEIGHT = "0px";
-      const MINIMUM_ITEMS_LIST_HEIGHT = "215px";
-      const MAXIMUM_ITEMS_LIST_HEIGHT = "270px";
-      const [itemsList, setItemsList] = useState(items);
-      const [viewAllLabel, setViewAllLabel] = useState("View all");
-      const [listHeight, setListHeight] = useState(MINIMUM_ITEMS_LIST_HEIGHT);
-      const [listItemsCount, setListItemsCount] = useState(defaultItemsVisible);
+      const listHeight =
+        listOverflow === "expand" ? "unset" : isSearchable ? "270px" : "215px";
 
-      // Separate effect for handling listOverflow "scroll"
-      React.useEffect(() => {
-        if (listOverflow === "scroll") {
-          setListHeight(MINIMUM_ITEMS_LIST_HEIGHT);
-          if (isSearchable) {
-            setListHeight(MAXIMUM_ITEMS_LIST_HEIGHT);
-          }
-          setItemsList(items);
-        }
-      }, [listOverflow, items, isSearchable]);
+      const isOverflowExpand =
+        items.length >= defaultItemsVisible && listOverflow === "expand";
+      const defaultItemsList = isOverflowExpand
+        ? items.slice(0, defaultItemsVisible)
+        : items;
+      const [itemsList, setItemsList] = useState(defaultItemsList);
+      const [isExpandable, setIsExpandable] = useState(true);
+
+      const selectedItemsCount: number = selectedItems[id]?.items.length || 0;
 
-      // Sets the selected items count on the menu button.
-      const getSelectedItemsCount: number =
-        selectedItems[id]?.items.length || 0;
+      const selectedItemsString = `item${selectedItemsCount === 1 ? "" : "s"}`;
+      const ariaLabelValue = `${buttonText}, ${selectedItemsCount} ${selectedItemsString} currently selected`;
+
+      // Get the styles for the component
       const styles = useMultiStyleConfig("MultiSelect", {
         isBlockElement,
-        width,
+        width: multiSelectWidth,
       });
 
+      // Only for storybook
```

Comment:
Again, let's not include fixes that only serve storybook in the code that will be published. We can add more examples in Storybook for the `listOverflow` values.

Reply:
Agree. I removed it. Did not add anything to storybook yet - want to see what we end up with when fixing the `useEffect` bug